src/integration.test.ts

```diff
@@ -58,65 +89,95 @@ describe("Integration Test", (): void => {
     });
   });
 
-  const mockedExec = async (load: boolean, command: string): Promise<void> => {
-    const stdout = "standard output from Bash command";
-    const stderr = "error output from Bash command";
-    child_process.exec.mockImplementationOnce(
-      (_command: any, _options: any, callback: any): any => {
-        callback(null, { stdout, stderr });
-      }
-    );
-
-    await (load ? docker.loadDockerImages() : docker.saveDockerImages());
-
-    expect(core.getInput).nthCalledWith(1, "key", { required: true });
-    expect(core.info).nthCalledWith<[string]>(1, command);
-    expect(child_process.exec).lastCalledWith(
+  const assertExecBashCommand = (
+    infoCallNum: number,
+    execCallNum: number,
+    command: string
+  ): void => {
+    expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
+    expect(child_process.exec).nthCalledWith(
+      execCallNum,
       command,
       EXEC_OPTIONS,
       expect.anything()
     );
-    expect(core.info).lastCalledWith(stdout);
-    expect(core.error).lastCalledWith(stderr);
+    expect(core.error).lastCalledWith(STDERR);
     expect(core.setFailed).not.toHaveBeenCalled();
   };
 
-  test("cache misses, then hits", async (): Promise<void> => {
-    // Attempt first cache restore.
-    await docker.loadDockerImages();
-
-    // Expect cache miss since cache has never been saved.
+  const assertLoadDockerImages = (cacheHit: boolean): void => {
+    /* TODO: refactor to share code across test files to enable cleaner decomposition without
+     * failing copy/paste linter.
+     */
+    /* jscpd:ignore-start */
     expect(core.getInput).nthCalledWith(1, "key", { required: true });
-    expect(core.setOutput).lastCalledWith(docker.CACHE_HIT, false);
+    expect(cache.restoreCache).lastCalledWith([docker.DOCKER_IMAGES_PATH], KEY);
+    expect(core.saveState).nthCalledWith(1, docker.CACHE_HIT, cacheHit);
+    expect(core.setOutput).lastCalledWith(docker.CACHE_HIT, cacheHit);
+    /* jscpd:ignore-end */
+    if (cacheHit) {
+      assertExecBashCommand(1, 1, loadCommand);
+    } else {
+      expect(core.info).nthCalledWith(
+        1,
+        "Recording preexisting Docker images. These include standard images pre-cached by GitHub Actions when Docker is run as root."
+      );
+      assertExecBashCommand(2, 1, LIST_COMMAND);
+      expect(core.info).nthCalledWith(3, dockerImages.join("\n"));
```

Comment:
It seems like the corresponding assert is missing after all other calls to `assertExecBashCommand`.